### PR TITLE
NEW: Handle ActionController::ParameterMissing for API requests

### PR DESCRIPTION
**Pull request by bugant**
Handle ActionController::ParameterMissing error for API requests and
return a bad request status with info about the missing parameter.

Related to issue #195.

### CODE REVIEW

File: test/functional/api/tomatoes_controller_test.rb [hunk 1]
```diff
@@ -91,6 +91,20 @@ class TomatoesControllerTest < ActionController::TestCase
       assert_match(/Must not overlap saved tomaotes/, parsed_response['base'].first)
     end
 
+    test 'POST /create, with bad parameters, it should return a bad request error' do
+      @user.tomatoes.create!
+
+      assert_no_difference('@user.tomatoes.size') do
+        # this request should fail because another tomato has been created
+        # less than 25 minutes ago
```
Comment (potomak) [inline test/functional/api/tomatoes_controller_test.rb]: This comment is wrong, isn't it?

File: test/functional/api/tomatoes_controller_test.rb [hunk 2]
```diff
@@ -91,6 +91,20 @@ class TomatoesControllerTest < ActionController::TestCase
...
+        post :create, token: '123', pachino: { tag_list: 'one, two' }
```
Comment (potomak) [inline test/functional/api/tomatoes_controller_test.rb]: 🍅

File: app/controllers/api/base_controller.rb [hunk 3]
```diff
@@ -2,6 +2,10 @@ module Api
   class BaseController < ActionController::Base
     protect_from_forgery with: :null_session
 
+    rescue_from(ActionController::ParameterMissing) do |err|
+      render json: {missing_param: err.param}, status: :bad_request
```
Comment (potomak) [inline app/controllers/api/base_controller.rb]: 👍  for the *symbolic* status.

`{ ... }` content should be padded by spaces. I think rubocop would complain about it, try to run it if you have time, not a blocker though.

File: test/functional/api/tomatoes_controller_test.rb (hunk 1 above)
Comment (bugant) [inline test/functional/api/tomatoes_controller_test.rb]: Yes, you are right @potomak (copy/paste error)

**Pull request merged**